### PR TITLE
refacto cache path fetching using scarb cache path

### PR DESCRIPTION
**Pull request by ametel01**
solves #11

### CODE REVIEW

**Review by mkaput — APPROVED** [review]
Looks good! We'll unfortunately have to merge this later, when Scarb containing this command will be released and widely used. I expect this to happen around Starknet 0.13

File: lib/cache-utils.js [hunk 1]
```diff
@@ -9,20 +9,16 @@ export const State = {
...
-  switch (platform) {
-    case "linux":
-      return path.join(home, ".cache/scarb");
-    case "darwin":
-      return path.join(home, `Library/Caches/com.swmansion.scarb`);
-    case "win32":
-      return path.join(process.env.APPDATA, "swmansion/scarb/config");
-    default:
-      throw new Error(`caching is not available on this platform: ${platform}`);
+export async function getCacheDirectory() {
```
Comment (mkaput) [inline lib/cache-utils.js:12]: Oh just caught this. Can you please add `await` to all invocations of this function?

**Review by mkaput — CHANGES_REQUESTED** [review]

File: lib/cache-utils.js (hunk 1 above)
Comment (mkaput) [inline lib/cache-utils.js:12]: Can you please add `await` before `getCacheDirectory` also here? [`await cache.saveCache([getCacheDirectory()], primaryKey);`](https://github.com/ametel01/setup-scarb/blob/2ba10e70db8028e48ca2f656dae04d2a782c4b73/lib/cache-save.js#L12) I believe it's needed, and currently we're caching by promise object address, not its computed value per se.

**Review by mkaput — CHANGES_REQUESTED** [review]

File: lib/cache-utils.js (hunk 1 above)
Comment (ametel01) [reply lib/cache-utils.js:12]: done

**Review by mkaput — APPROVED** [review]
❤️❤️❤️❤️❤️

**Comment by mkaput** [comment]
OK, I gave this a thought and I think we have to preserve the old behaviour indefinitely, if we want to stick with `v1` version line for this action. I'll merge this PR now, and I'll add falling back logic in separate PR.

**Pull request merged**